ui/src/app/base/components/TableActionsDropdown/TableActionsDropdown.tsx

```diff
@@ -0,0 +1,48 @@
+import type { Props as ButtonProps } from "@canonical/react-components/dist/components/Button";
+
+import TableMenu from "app/base/components/TableMenu";
+
+export type TableAction<A> = {
+  label: string;
+  show?: boolean;
+  type: A;
+};
+
+// This allows the "data-test" attribute to be used for the action links, which
+// is technically not a valid HTML prop, but we use it throughout maas-ui.
+type TableActionsLink = ButtonProps & { "data-test": string };
```

Comment:
This seems strange as it seems to work for other types that we've defined. Have we done something wrong with the Button type? Maybe `HTMLProps<HTMLElement>` doesn't include `data-*`.

Reply:
Hmm it doesn't complain if you add use it directly on a button `<Button data-test="works" />`.

Reply:
I think the issue might lie with `ContextualMenu`'s `MenuLink` type, but I can't quite figure it out...